bottlecap/src/logs/agent.rs

```diff
@@ -20,10 +21,15 @@ pub struct LogsAgent {
 
 impl LogsAgent {
     #[must_use]
-    pub fn run(function_arn: &str, datadog_config: Arc<config::Config>) -> LogsAgent {
+    pub fn run(
+        function_arn: &str,
+        tags_provider: Arc<provider::Provider>,
+        datadog_config: Arc<config::Config>,
+    ) -> LogsAgent {
         let function_arn = function_arn.to_string();
         let aggregator: Arc<Mutex<Aggregator>> = Arc::new(Mutex::new(Aggregator::default()));
-        let mut processor: Processor = Processor::new(function_arn, Arc::clone(&datadog_config));
+        let mut processor: Processor =
+            Processor::new(function_arn, tags_provider, Arc::clone(&datadog_config));
```

Comment:
If the `tags_provider` contains the function_arn, should we even be sending it now?

Reply:
Or is it too hard since the tags provider only gives us the complete set?

Reply:
yeah technically we're duplicating here, but the tags_provider uses the function ARN for it's own purposes and immediately carves it up, so it's not meant to be the canonical holder of function ARN.

We should refactor the logs agent to process this from a metadata hashmap (like we did with tag providers) so that we're not coupled to Lambda though.

Reply:
Sounds good